.github/workflows/test.yml

```diff
@@ -26,21 +25,24 @@ jobs:
       - name: Checkout the repo
         uses: actions/checkout@v3
 
+      - name: Install poetry
+        run: pipx install poetry
+
+      - name: Configure poetry
+        run: poetry config virtualenvs.in-project true
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python-version }}
-
-      - name: Install poetry
-        run: pipx install poetry
+          cache: 'poetry'
```

Comment:
Using a cache like this really helped. A `Test` workflow run w/o the cache took 3 minutes while the same run w/ the cache took 2 minutes. This caching feature was added to the other workflows as well.